component/common/src/main/java/org/wso2/carbon/extension/identity/emailotp/common/util/OneTimePasswordUtils.java

```diff
@@ -0,0 +1,249 @@
+/*
+ * Copyright (c) 2021, WSO2 Inc. (http://www.wso2.com).
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.extension.identity.emailotp.common.util;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.extension.identity.emailotp.common.constant.Constants;
+import org.wso2.carbon.extension.identity.emailotp.common.exception.EmailOtpServerException;
+
+import java.security.InvalidKeyException;
+import java.security.NoSuchAlgorithmException;
+import java.security.SecureRandom;
+
+import javax.crypto.Mac;
+import javax.crypto.spec.SecretKeySpec;
+
+/**
+ * OTP generation utils.
+ * <p>
+ * Important: This class duplicates OTP util methods of the authenticator
+ * module's 'org.wso2.carbon.identity.authenticator.emailotp.OneTimePassword' class.
+ * Any fixes here, should be reflected there as well.
+ */
+public class OneTimePasswordUtils {
+
+    // Each OTP digit(0-9) has a corresponding checksum digit as in the below array.
+    private static final int[] CHECKSUM_DIGITS = {0, 2, 4, 6, 8, 1, 3, 5, 7, 9};
+
+    private static final Log log = LogFactory.getLog(OneTimePasswordUtils.class);
+
+    public static String getRandomNumber(int size) {
+
+        StringBuilder generatedToken = new StringBuilder();
+        try {
+            SecureRandom number = SecureRandom.getInstance(Constants.ALGORITHM_NAME);
+            // Generate 20 integers 0..20
+            for (int i = 0; i < size; i++) {
+                generatedToken.append(number.nextInt(9));
+            }
+        } catch (NoSuchAlgorithmException e) {
+            log.error("Unable to find the Algorithm.", e);
+        }
+
+        return generatedToken.toString();
+    }
+
+    /**
+     * @param num    The number to calculate the checksum for.
```

Comment:
Missing method description. Shall we complete the comment with a description?

Reply:
Addressed in 148a8f957bdd2c4d574fb53e81284dfc017db263